core/base/ftmTree/FTMTreeUtils.h

```diff
@@ -0,0 +1,158 @@
+/// \ingroup base
+/// \class FTMTreeUtils
+/// \author Mathieu Pont (mathieu.pont@lip6.fr)
+/// \date 2021.
+///
+/// Utils function for manipulating FTMTree class
+
+#ifndef _FTMTREEUTILS_H
+#define _FTMTREEUTILS_H
+
+#pragma once
+
+#include <FTMTree.h>
+
+namespace ttk {
+  namespace ftm {
+
+    struct MergeTree {
+      ftm::Scalars scalars;
+      ftm::Params params;
+      ftm::FTMTree_MT tree;
+      MergeTree(ftm::Scalars scalarsT, ftm::Params paramsT)
+        : scalars(scalarsT), params(paramsT),
+          tree(&params, &scalars, params.treeType) {
+        tree.makeAlloc();
+      }
+    };
+
+    // --------------------
+    // Is
+    // --------------------
+
+    bool isNodeOriginDefined(FTMTree_MT *tree, idNode nodeId);
```

Comment:
why having these methods as standalone (with the tree as parameters) instead of putting them in the merge tree class (or in a inherited one) ?

Reply:
(this could be used in the future by the FTM wrapper to display persistence related info for example)

Reply:
that's a good call. @MatPont could you have a look at that?

Reply:
Indeed, it will be fixed in the next commit !

Reply:
Ok, so I've pushed a commit that solve this problem. The FTMTreeUtils functions are now class functions of the FTMTree_MT class. 
I've replaced the original getParent function with my version that tests before accessing the first up super arc of a node if it exists. If it is a problem tell me!
Thanks!

Reply:
Your `getParent` method is cleaner as the check may avoid bugs, it may have a negative impact one the perf.
In terms of architecture, I do not like the `#include` in the middle of the class definition. Could you add your method here instead ? (sorry for the move)

Reply:
For the getParent we can keep both versions if you prefer, I can rename the new version getParentSafe or something like that.
For the include, indeed, it was not quite a good way to do. I did it to avoid to make too much longer FTMTree_MT.h (especially for a next pull request in which I will add even more utils functions). But I will remove the include and add the definitions here like you suggested.
It will be done in the next commit.
Thanks!

Reply:
Ok, so I've pushed a commit that rename the new getParent to getParentSafe and to fix the include in the middle of the class definition!

Reply:
@MatPont regarding your function `getParentSafe`: 
in ttk, we use the flag `TTK_ENABLE_KAMIKAZE` to skip safeguards. 
a good example is `core/base/triangulation/Triangulation.h`
you'll see there that all the sanity checks are in between `#ifndef TTK_ENABLE_KAMIKAZE`
this flag is meant to be used in development mode, but not in production (to get the best performance).
so if you're confident that this function is never called on the root in the ttk-data examples, you should probably put the if comment between one of these `#ifndef` blocks.